pkg/controller/istiocontrolplane/listenersv2.go

```diff
@@ -0,0 +1,63 @@
+package istiocontrolplane
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"istio.io/operator/pkg/apis/istio/v1alpha2"
+	"istio.io/operator/pkg/helmreconciler"
+)
+
+// IstioStatusUpdater is a RenderingListener that updates the status field on the IstioControlPlane
+// instance based on the results of the Reconcile operation.
+type IstioStatusUpdaterV2 struct {
+	*helmreconciler.DefaultRenderingListener
+	instance   *v1alpha2.IstioControlPlane
+	reconciler *helmreconciler.ISCPReconciler
+}
+
+// NewIstioRenderingListener returns a new IstioRenderingListener, which is a composite that includes IstioStatusUpdater
+// and IstioChartCustomizerListener.
+func NewIstioRenderingListenerV2(instance *v1alpha2.IstioControlPlane) *IstioRenderingListener {
+	return &IstioRenderingListener{
+		&helmreconciler.CompositeRenderingListener{
+			Listeners: []helmreconciler.RenderingListener{
+				NewChartCustomizerListener(),
+				NewIstioStatusUpdaterV2(instance),
+			},
+		},
+	}
+}
+
+// NewIstioStatusUpdater returns a new IstioStatusUpdater instance for the specified IstioControlPlane
+func NewIstioStatusUpdaterV2(instance *v1alpha2.IstioControlPlane) helmreconciler.RenderingListener {
+	return &IstioStatusUpdaterV2{
+		DefaultRenderingListener: &helmreconciler.DefaultRenderingListener{},
+		instance:                 instance,
+	}
+}
+
+// EndReconcile updates the status field on the IstioControlPlane instance based on the resulting err parameter.
+//TODO
```

Comment:
Could you state here what is still todo?

Reply:
ok that's something for my personal reminder to check this part of logic. would remove at the end